sycl/plugins/level_zero/pi_level_zero.hpp

```diff
@@ -295,15 +298,17 @@ struct _pi_context : _pi_object {
 const pi_uint32 DynamicBatchStartSize = 4;
 
 struct _pi_queue : _pi_object {
-  _pi_queue(ze_command_queue_handle_t Queue, pi_context Context,
+  _pi_queue(ze_command_queue_handle_t Queue,
+            ze_command_queue_handle_t CopyQueue, pi_context Context,
             pi_device Device, pi_uint32 BatchSize)
-      : ZeCommandQueue{Queue}, Context{Context}, Device{Device},
+      : ZeComputeCommandQueue{Queue},
+        ZeCopyCommandQueue{CopyQueue}, Context{Context}, Device{Device},
         QueueBatchSize{BatchSize > 0 ? BatchSize : DynamicBatchStartSize},
         UseDynamicBatching{BatchSize == 0} {}
 
   // Level Zero command queue handle.
-  ze_command_queue_handle_t ZeCommandQueue;
-
+  ze_command_queue_handle_t ZeComputeCommandQueue;
+  ze_command_queue_handle_t ZeCopyCommandQueue;
```

Comment:
I believe that dividing the commands stream into 2 queues creates serious synchronization issues for in-order queue. Now the plugin needs to do something more to ensure that commands are executed one after another, and this something is crossing a Level-Zero queue boundary.

Tag @rbegam for awareness

Reply:
I think rather than relying on the underlying run-times for ensuring in-order queues, that we should do one of two different things.
1 - We could just have the DPC++ run-time create an dependence of on each submission with the previous submission.  This is easily implemnted by having the RT place a wait on event from the previous submission on the current submission.
2 - The level zero plugin can, for each queue, have a field in the queue that is the event for the previously created event, and it can implicitly add that to the wait on event list when doing a submission.
I think either of these methods is probably better than relying on the underlying level zero runtime queues inorder properties.

Reply:
Yes, that's what I meant, we need to add [many] more dependencies for these 2 queues to truly implement in-order semantics, when required. I do think we have to continue rely in what are the underlying runtimes guarantee, since that it is going to remove the redundant resource/performance overhead. For example, OpenCL RT does support in-order queues internally, so we don't want to do (1). Level-Zero RT guarantees in-order execution of commands within the same command-list, and we'd use that when implement something like you describe in (2) to minimize on events.

I think that we have to address this issue in this PR since this is breaking current in-order queues behavior. I'd start by just disabling use of a blitter-engines for in-order SYCL queues, and add proper sync separately.

Reply:
We have a resolution which will need to be in sync with the overall effort to support in-order queues in level zero. Under discussion.

Reply:
We have a resolution that will temporarily disable blitter engine use for in-order queues. Need some discussion with owner of in-order queue supporting effort.

Reply:
Are you going to add disabling for in-order queue?

Reply:
Added, Please see Line 572 and 653.